src/lib/quoteEngine.js

```diff
@@ -78,79 +157,88 @@ const quoteEngine = (store, productId, amount, period, coverAsset) => {
   const amountToAllocate = divCeil(coverAmountInNxm, NXM_PER_ALLOCATION_UNIT).mul(NXM_PER_ALLOCATION_UNIT);
   console.log('Amount to allocate:', formatEther(amountToAllocate), 'nxm');
 
-  const zeroPool = {
-    poolId: 0,
-    premiumInNxm: MaxUint256,
-    premiumInAsset: MaxUint256,
-    coverAmountInNxm: MaxUint256,
-    coverAmountInAsset: MaxUint256,
-  };
-
-  const poolsWithPremium = productPools.map(pool => {
+  const poolsData = productPools.map(pool => {
     const { poolId, targetPrice, bumpedPrice, bumpedPriceUpdateTime, allocations, trancheCapacities } = pool;
 
     const totalCapacity = trancheCapacities
       .slice(firstUsableTrancheIndex)
       .reduce((total, capacity) => total.add(capacity), Zero)
       .mul(NXM_PER_ALLOCATION_UNIT);
 
-    const initiallyUsedCapacity = allocations
+    const initialCapacityUsed = allocations
       .slice(firstUsableTrancheIndex)
       .reduce((total, allocation) => total.add(allocation), Zero)
       .mul(NXM_PER_ALLOCATION_UNIT);
 
-    const capacityInNxm = totalCapacity.sub(initiallyUsedCapacity);
-    const capacity = Object.entries(assetRates).map(([assetId, rate]) => ({
-      assetId,
-      amount: capacityInNxm.mul(rate).div(WeiPerEther),
-    }));
-
-    console.log('Pool:', poolId);
-    console.log('Initially used capacity:', formatEther(initiallyUsedCapacity), 'nxm');
-    console.log('Total pool capacity    :', formatEther(totalCapacity), 'nxm');
-    console.log('Pool capacity          :', formatEther(capacityInNxm), 'nxm');
-
-    if (initiallyUsedCapacity.add(amountToAllocate).gt(totalCapacity)) {
-      return { ...zeroPool, poolId };
-    }
-
-    // TODO: account for global min price
     const basePrice = product.useFixedPrice
       ? targetPrice
       : calculateBasePrice(targetPrice, bumpedPrice, bumpedPriceUpdateTime, now);
 
+    return {
+      poolId,
+      basePrice,
+      initialCapacityUsed,
+      totalCapacity,
+    };
+  });
+
+  const { assets } = store.getState();
+  const daiRate = assetRates[assets.DAI];
+  const minUnitSizeInNxm = MIN_UNIT_SIZE_DAI.mul(WeiPerEther).div(daiRate);
```

Comment:
note we are adding dynamic computation of unit size in NXM based on the fixed value of 10.000 DAI